s3fs/mapping.py

```diff
@@ -65,24 +63,16 @@ def __setitem__(self, key, value):
             f.write(value)
 
     def keys(self):
-        return iter(map(lambda x: x[len(self.root) + 1:], self.s3.walk(self.root)))
+        return (x[len(self.root) + 1:] for x in self.s3.walk(self.root))
```

Comment:
The failure must be here: the generator is iterable but not an interator (but gets called by __iter__)